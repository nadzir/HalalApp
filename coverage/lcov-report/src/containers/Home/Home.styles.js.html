<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/containers/Home/Home.styles.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/containers/Home</a> Home.styles.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/1</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { StyleSheet } from 'react-native'
import { COLOURS } from '../../../config/constants'
&nbsp;
export const styles = StyleSheet.create({
  outerViewStyle: {
    flexDirection: 'column',
    backgroundColor: COLOURS.HOME_BACKGROUND,
    flex: 1,
    justifyContent: 'space-between'
  },
  topTextStyle: {
    color: COLOURS.BRAND,
    marginTop: 100,
    height: '50%',
    width: '80%',
    textAlign: 'left'
  },
  bottomTextStyle: {
    color: COLOURS.BRAND,
    width: '80%',
    left: 0,
    margin: 20,
    textAlign: 'left'
  }
})
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jun 16 2018 20:46:23 GMT+0800 (+08)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
