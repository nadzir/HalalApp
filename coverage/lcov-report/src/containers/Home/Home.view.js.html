<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/containers/Home/Home.view.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/containers/Home</a> Home.view.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>14/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>14/14</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">500x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react'
import { View, TouchableOpacity, Animated, Easing } from 'react-native'
import { Text } from 'react-native-elements'
import { Actions } from 'react-native-router-flux'
import { styles } from '../Home'
import { material, systemWeights } from 'react-native-typography'
import { analytics } from '../../analytics'
&nbsp;
const arr = []
for (var i = 0; i &lt; 500; i++) {
  arr.push(i)
}
&nbsp;
export class HomeView extends Component {
  constructor () {
    super()
    this.titleAnimatedOpacity = new Animated.Value(0)
    this.textAnimatedOpacity = new Animated.Value(0)
  }
&nbsp;
  componentDidMount () {
    this.animate()
    analytics.logEvent('screen', {'name': 'Home'})
    analytics.setCurrentScreen('Home')
  }
&nbsp;
  createAnimation (value, duration, easing, delay = 0) {
    return Animated.timing(
      value,
      {
        toValue: 1,
        duration,
        easing,
        delay
      }
    )
  }
&nbsp;
  animate () {
    this.titleAnimatedOpacity.setValue(0)
    this.textAnimatedOpacity.setValue(0)
&nbsp;
    Animated.parallel([
      this.createAnimation(this.titleAnimatedOpacity, 1000, Easing.bounce),
      this.createAnimation(this.textAnimatedOpacity, 2000, Easing.bounce)
    ]).start()
  }
&nbsp;
  render () {
    return (
      &lt;View style={styles.outerViewStyle}&gt;
        &lt;TouchableOpacity
          activeOpacity={0.5}
          onPress={Actions.camera}
        &gt;
          &lt;Animated.View
            style={{
              opacity: this.titleAnimatedOpacity,
              alignItems: 'center'}}&gt;
            &lt;Text
              style={[material.display3, systemWeights.light, styles.topTextStyle]} &gt;
              Is This {'\n'}
              Halal?
            &lt;/Text&gt;
          &lt;/Animated.View&gt;
          &lt;Animated.View
            style={{
              opacity: this.textAnimatedOpacity,
              alignItems: 'center'}}&gt;
            &lt;Text style={[material.headline, systemWeights.light, styles.bottomTextStyle]} h4&gt;
               Try taking a photo{'\n'}
               of a brand{'\n'}
               or a logo{'\n'}
               and check if it is halal.
            &lt;/Text&gt;
&nbsp;
          &lt;/Animated.View&gt;
        &lt;/TouchableOpacity&gt;
      &lt;/View&gt;
    )
  }
}
&nbsp;
export default HomeView
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jun 16 2018 20:46:23 GMT+0800 (+08)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
