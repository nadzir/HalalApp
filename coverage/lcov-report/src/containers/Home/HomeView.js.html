<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/containers/Home/HomeView.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/containers/Home</a> HomeView.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import { View } from 'react-native'
import { SocialIcon, Text } from 'react-native-elements'
import { Actions } from 'react-native-router-flux'
import { COLOURS } from '../../../config/constants'
&nbsp;
const outerViewStyle = {
  flexDirection: 'column',
  backgroundColor: COLOURS.HOME_BACKGROUND,
  flex: 1,
  justifyContent: 'center'
}
&nbsp;
const textStyle = {color: COLOURS.WHITE, textAlign: 'center'}
&nbsp;
export const HomeView = () =&gt; (
  &lt;View style={outerViewStyle}&gt;
    &lt;View style={{height: 100}}&gt;
      &lt;Text style={textStyle} h1&gt;HalalGo&lt;/Text&gt;
    &lt;/View&gt;
    &lt;View style={{height: 100}}&gt;
      &lt;Text style={textStyle} h3&gt;One Stop Halal Guide&lt;/Text&gt;
    &lt;/View&gt;
    &lt;View &gt;
      &lt;SocialIcon
        title='Sign In With Facebook'
        button
        type='facebook'
        onPress={Actions.halalFinder}
      /&gt;
    &lt;/View&gt;
  &lt;/View&gt;
)
&nbsp;
export default HomeView
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Apr 21 2018 08:59:07 GMT+0800 (+08)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
