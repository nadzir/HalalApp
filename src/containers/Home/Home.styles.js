import { StyleSheet } from 'react-native'
import { COLOURS } from '../../../config/constants'

export const styles = StyleSheet.create({
  outerViewStyle: {
    flexDirection: 'column',
    backgroundColor: 'white',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  avatar: {
    margin: 20,
    backgroundColor: COLOURS.BRAND
  },
  topTextStyle: {
    color: COLOURS.BRAND,
    textAlign: 'center'
  },
  bottomTextStyle: {
    color: COLOURS.BRAND_COMPLIMENT
  }
})
