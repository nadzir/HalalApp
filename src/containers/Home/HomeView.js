import React from 'react'
import { View } from 'react-native'
import { SocialIcon, Text } from 'react-native-elements'
import { Actions } from 'react-native-router-flux'
import { COLOURS } from '../../../config/constants'

const outerViewStyle = {
  flexDirection: 'column',
  backgroundColor: COLOURS.HOME_BACKGROUND,
  flex: 1,
  justifyContent: 'center'
}

const textStyle = {color: COLOURS.WHITE, textAlign: 'center'}

export const HomeView = () => (
  <View style={outerViewStyle}>
    <View style={{height: 100}}>
      <Text style={textStyle} h1>HalalGo</Text>
    </View>
    <View style={{height: 100}}>
      <Text style={textStyle} h3>One Stop Halal Guide</Text>
    </View>
    <View >
      <SocialIcon
        title='Sign In With Facebook'
        button
        type='facebook'
        onPress={Actions.halalFinder}
      />
    </View>
  </View>
)

export default HomeView
