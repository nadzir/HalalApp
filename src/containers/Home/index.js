export * from './Home.view'
export * from './Home.styles'
