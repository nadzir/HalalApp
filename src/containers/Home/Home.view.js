import React, { Component } from 'react'
import { View, TouchableOpacity, Animated, Easing } from 'react-native'
import { Text } from 'react-native-elements'
import { Actions } from 'react-native-router-flux'
import { styles } from '../Home'
import { material, systemWeights } from 'react-native-typography'
import { analytics } from '../../analytics'

const arr = []
for (var i = 0; i < 500; i++) {
  arr.push(i)
}

export class HomeView extends Component {
  constructor () {
    super()
    this.titleAnimatedOpacity = new Animated.Value(0)
    this.textAnimatedOpacity = new Animated.Value(0)
  }

  componentDidMount () {
    this.animate()
    analytics.logEvent('screen', {'name': 'Home'})
    analytics.setCurrentScreen('Home')
  }

  createAnimation (value, duration, easing, delay = 0) {
    return Animated.timing(
      value,
      {
        toValue: 1,
        duration,
        easing,
        delay
      }
    )
  }

  animate () {
    this.titleAnimatedOpacity.setValue(0)
    this.textAnimatedOpacity.setValue(0)

    Animated.parallel([
      this.createAnimation(this.titleAnimatedOpacity, 1000, Easing.bounce),
      this.createAnimation(this.textAnimatedOpacity, 2000, Easing.bounce)
    ]).start()
  }

  render () {
    return (
      <View style={styles.outerViewStyle}>
        <TouchableOpacity
          activeOpacity={0.5}
          onPress={Actions.camera}
        >
          <Animated.View
            style={{
              opacity: this.titleAnimatedOpacity,
              alignItems: 'center'}}>
            <Text
              style={[material.display3, systemWeights.light, styles.topTextStyle]} >
              Is This {'\n'}
              Halal?
            </Text>
          </Animated.View>
          <Animated.View
            style={{
              opacity: this.textAnimatedOpacity,
              alignItems: 'center'}}>
            <Text style={[material.headline, systemWeights.light, styles.bottomTextStyle]} h4>
               Try taking a photo{'\n'}
               of a brand{'\n'}
               or a logo{'\n'}
               and check if it is halal.test
            </Text>

          </Animated.View>
        </TouchableOpacity>
      </View>
    )
  }
}

export default HomeView
