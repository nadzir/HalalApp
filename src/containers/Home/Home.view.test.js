import React from 'react'
import { HomeView } from '../Home'

import renderer from 'react-test-renderer'

jest.mock('../../analytics', () => {
  return {
    analytics: {
      logEvent: () => {},
      setCurrentScreen: () => {}
    }
  }
})

test('renders correctly', () => {
  const tree = renderer.create(<HomeView />).toJSON()
  expect(tree).toMatchSnapshot()
})
