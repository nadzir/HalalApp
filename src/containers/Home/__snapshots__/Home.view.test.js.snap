// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F2F1EF",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "alignItems": "center",
          "opacity": 0,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0,0,0,0.54)",
                "fontFamily": "System",
                "fontSize": 56,
                "fontWeight": "400",
                "letterSpacing": 0.273438,
                "lineHeight": 64,
              },
              Object {
                "backgroundColor": "transparent",
                "fontFamily": "System",
                "fontWeight": "300",
              },
              Object {
                "color": "#26A65B",
                "height": "50%",
                "marginTop": 100,
                "textAlign": "left",
                "width": "80%",
              },
            ],
          ]
        }
      >
        Is This 
        

        Halal?
      </Text>
    </View>
    <View
      collapsable={undefined}
      style={
        Object {
          "alignItems": "center",
          "opacity": 0,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0,0,0,0.87)",
                "fontFamily": "System",
                "fontSize": 24,
                "fontWeight": "400",
                "letterSpacing": 0.351562,
                "lineHeight": 32,
              },
              Object {
                "backgroundColor": "transparent",
                "fontFamily": "System",
                "fontWeight": "300",
              },
              Object {
                "color": "#26A65B",
                "left": 0,
                "margin": 20,
                "textAlign": "left",
                "width": "80%",
              },
            ],
          ]
        }
      >
        Try taking a photo
        

        of a brand
        

        or a logo
        

        and check if it is halal.
      </Text>
    </View>
  </View>
</View>
`;
